src/wallet/wallet.cpp

```diff
@@ -55,6 +55,9 @@ const std::map<uint64_t,std::string> WALLET_FLAG_CAVEATS{
         "destinations in the past. Until this is done, some destinations may "
         "be considered unused, even if the opposite is the case."
     },
+    {WALLET_FLAG_EXTERNAL_SIGNER,
+        "The ability to toggle this flag may be removed in a future update."
```

Comment:
Why do you expect this to be removed in future?

Reply:
If we ever need to make a backwards incompatible change to these types of wallets.

I don't have anything specific in mind. Right now it's perfectly safe to treat such a wallet as a regular watch-only wallet, create a PSBT and call HWI yourself. All unknown future fields and flag are simply ignored (and not deleted).

But perhaps there's some future fancy multisig setup with radioactive nonces and hash pre-images, that would make any naive use of the wallet keys dangerous.

Reply:
Likely possibility: In the future, setting it may require providing a specific path to the signer program